### ISSUE TITLE
Commandline argparser of DataSelectionApplet swallows parameters with spaces

### ISSUE THREAD

**Issue by chaubold** [ilastik/ilastik#1422]
For the upcoming [Tracking Export Plugin system](https://github.com/ilastik/ilastik/tree/tracking-export-plugin-system) I am currently adding batch processing capabilities so that @JaimeIvanCervantes can use it for his next steps.

Unfortunately, the `unspecified_input_files` parameter to the argument parser in [dataSelectionApplet.py#L115](https://github.com/ilastik/ilastik/blob/master/ilastik/applets/dataSelection/dataSelectionApplet.py#L115) somehow takes in other parameters **if they contain spaces**, and then complains about unconfigured `raw_data` and `segmentation_image`.

```sh
./run_ilastik.sh --project .../test.ilp --headless --raw_data .../trackingVideoWithMergersAndDivisions.h5/data --segmentation_image .../trackingVideoWithMergersAndDivisionsSegmentation.h5/exported_data --export_source="Plugin" --export_plugin="H5 Event Sequence"

...
RuntimeError: Could not create instance of '<class 'ilastik.workflows.tracking.conservation.conservationTrackingWorkflow.ConservationTrackingWorkflowFromBinary'>'
args   = (<ilastik.shell.headless.headlessShell.HeadlessShell object at 0x11d3b9790>, True, ['--raw_data', '/Users/chaubold/hci/code/embryonic/tests/boxesTestDataset/trackingVideoWithMergersAndDivisions.h5/data', '--segmentation_image', '/Users/chaubold/hci/code/embryonic/tests/boxesTestDataset/trackingVideoWithMergersAndDivisionsSegmentation.h5/exported_data', '--export_source=Plugin', '--export_plugin=H5 Event Sequence'], [])
kwargs = {}
The exception was:
Invalid command line arguments: All roles must be configured explicitly.
Use the following flags to specify which files are matched with which inputs:
--raw_data, --segmentation_image
```

And when inspecting the line above with the debugger, I see that
```py
unspecified_input_files = <type 'list'>: ['--export_plugin=H5 Event Sequence']
unused_args = <type 'list'>: ['--export_source=Plugin']
```
which doesn't make any sense. Changing the order of the command line arguments doesn't work either. 

But **if I change the plugin names to contain no spaces, everything works fine**!

@JaimeIvanCervantes I think that is why you changed the export source names in the Tracking workflow to use dashes instead of spaces, right?

@stuarteberg Do you know any less restrictive workaround than not using spaces in any parameters? I can fix the problem for now by not using spaces in plugin names, but this was quite hard to narrow down, and we should make it less confusing for other developers/users.

**Comment by JaimeIvanCervantes** [comment]
Yes, that's why I'm using dashes eg: `Table-Only` to export the csv table.

**Comment by k-dominik** [comment]
Just FYI, this is a know python issue: https://bugs.python.org/issue22433

So this only shows up if you don't use standard datasets (like you would in pixel classification). In tracking you have to specify input data roles, so that's where the bug finds its way.

For the time being I will update the docs and all help strings to contain no spaces.
